content/academy/puppeteer_playwright/page/interacting_with_a_page.md

```diff
@@ -0,0 +1,170 @@
+2. Type **hello world** into the search bar
+3. Press **Enter**
+4. Wait for the results page to load
+5. Click on the first result
+6. Read the title of the result's page
+7. Screenshot the page
+
+Though it seems complex, the wonderful **Page** API makes all of these actions extremely easy to perform.
+
+## [](#clicking-and-pressing-keys) Clicking & pressing keys
+
+Let's first focus on the first 3 steps listed above. By using `page.click()` and the CSS selector of the element to click, we can click an element:
+
+```marked-tabs
+<marked-tab header="Playwright" lang="JavaScript">
+// Click the "I agree" button
+await page.click('button:has-text("I agree")');
+</marked-tab>
+<marked-tab header="Puppeteer" lang="JavaScript">
+// Click the "I agree" button
+await page.click('button + button:nth-child(2)');
+</marked-tab>
+
+> Notice that in the Playwright example, we are using a different selector than in the Puppeteer example. This is because Playwright supports [many custom CSS selectors](https://playwright.dev/docs/selectors#text-selector), such as the **has-text** pseudo class.
+
+Then, we can type some text into an input field with `page.type()`; passing a CSS selector as the first, and the string to input as the second parameter:
+
+```JavaScript
+// Type the query into the search box
+await page.type('input[title="Search"]', 'hello world');
+```
+
+Finally, we can press a single key by accessing the `keyboard` property of `page` and calling the `press()` function on it:
+
+```JavaScript
+// Press enter
+await page.keyboard.press('Enter');
+```
+
+So far, we've got this:
+
+```marked-tabs
+<marked-tab header="Playwright" lang="javascript">
+import { chromium } from 'playwright';
+
+const browser = await chromium.launch({ headless: false });
+
+const page = await browser.newPage();
+
+await page.goto('https://google.com/');
+
+// Click the "I agree" button
+await page.click('button:has-text("I agree")');
+
+// Type the query into the search box
+await page.type('input[title="Search"]', 'hello world');
+
+// Press enter
+await page.keyboard.press('Enter');
+
+await page.waitForTimeout(10000)
+await browser.close();
+</marked-tab>
+<marked-tab header="Puppeteer" lang="javascript">
+import puppeteer from 'puppeteer';
+
+const browser = await puppeteer.launch({ headless: false });
+
+const page = await browser.newPage();
+
+await page.goto('https://google.com/');
+
+// Click the "I agree" button
+await page.click('button + button:nth-child(2)');
+
+// Type the query into the search box
+await page.type('input[title="Search"]', 'hello world');
+
+// Press enter
+await page.keyboard.press('Enter');
+
+await page.waitForTimeout(10000)
+await browser.close();
+</marked-tab>
+```
+
+When we run it, we leave off on the results page:
+
+![Google results page reached by headless browser]({{@asset puppeteer_playwright/page/images/google-results.webp}})
+
+Great! So now all we have to do is click the first result, which matches the selector `.g a`:
```

Comment:
```suggestion
Great! So now all we have to do is click the first result, which matches the CSS selector `.g a`:
```

Reply:
I would explain that it will click the first element found with a `.g a` selector. Because there are many elements like that on the pag
<img width="327" alt="CleanShot 2022-05-14 at 22 25 29@2x" src="https://user-images.githubusercontent.com/23726914/168447221-9f0e1de4-a52e-42d4-b93b-d7a22b4f5f12.png">
e